requestor-tutorials/service-development/service-example-6-external-api-request.md

```diff
@@ -0,0 +1,183 @@
+
+Following example will show cases 2. and 3. so it will be necesary to start a [local instance of a Provider](../../provider-tutorials/provider-tutorial.md).
+
+## Example app
+
+Example app will make a request to an external API using Provider's network and then it will print API response to the console.
+
+### 1. Manifest file
+
+In order for an app to make an _Outbound Network_ request it needs to declare which tools it will use and which URLs it will access in a [Computation Payload Manifest](../vm-runtime/computation-payload-manifest.md).
+
+Our example will make a HTTPS request using `curl` to a public REST API with URL `https://api.coingecko.com`.
+
+_Computation Payload Manifest_ will need to have following objects:
+
+- [`net`](../vm-runtime/computation-payload-manifest.md#compmanifestnet--object) computation constraints with `URL`s the app will access (`https://api.coingecko.com`)
+- [`script`](../vm-runtime/computation-payload-manifest.md#compmanifestscript) computation constraint with `command`s app will execute (`curl`)
+- [`payload`](../vm-runtime/computation-payload-manifest.md#payload-object) defining [Golem image](../vm-runtime#preparing-a-vm-image) containing tools used by the app (`curl`)
+
+Example _Computation Payload Manifest_ must follow a specific [schema](../vm-runtime/computation-payload-manifest.md#manifest-schema), and for our example it will take form of following `manifest.json` file:
+
+```json
+{
+  "version": "0.1.0",
+  "createdAt": "2022-07-26T12:51:00.000000Z",
+  "expiresAt": "2100-01-01T00:01:00.000000Z",
+  "metadata": {
+    "name": "External API call example",
+    "description": "Example manifest of a service making an outbound call to the external API",
+    "version": "0.1.0"
+  },
+  "payload": [
+    {
+      "platform": {
+        "arch": "x86_64",
+        "os": "linux"
+      },
+      "urls": [
+        "http://yacn2.dev.golem.network:8000/docker-golem-script-curl-latest-d75268e752.gvmi"
+      ],
+      "hash": "sha3:e5f5ddfd649525dbe25d93d9ed51d1bdd0849933d9a5720adb4b5810"
+    }
+  ],
+  "compManifest": {
+    "version": "0.1.0",
+    "script": {
+      "commands": [
+        "run .*curl.*",
+      ],
+      "match": "regex"
+    },
+    "net": {
+      "inet": {
+        "out": {
+          "protocols": ["https"],
+          "urls": ["https://api.coingecko.com"]
+        }
+      }
+    }
+  }
+}
+```
+
+Created file should be [verified using JSON schema](../vm-runtime/computation-payload-manifest.md#schema-verification).
+
+Then it needs to be encoded in `base64`:
+
+```sh
+ base64 --wrap=0 manifest.json.base64 > manifest.json.base64
+```
+### 2. Yapapi example app
+
+Base64 encoded _manifest_ can be configured using [`yapapi.payload.vm.manifest`](https://yapapi.readthedocs.io/en/latest/api.html#module-yapapi.payload.manifest) function resulting in following `externa_api_request.py` file:
```

Comment:
Relevant [yapapi PR](https://github.com/golemfactory/yapapi/pull/1008/files#diff-65869ccf135362e10237d1c22b5659f99db8c10d6bebaa1e51919f3f2626fd93R114) adding documentation to this function got created.